### DISCUSSION TITLE
Github Action Docker build error | long command

### DISCUSSION THREAD

**Discussion by serdarjan1995** [community/community#129532] [Actions] [answered]
### Github Action error

Question

### Body

I have a following job on Github actions
```
- name: Build api Docker image
  run: >
    docker build --tag api:latest
    --build-arg MONGODB_URL=${{ secrets.MONGODB_URL }}
    --build-arg OPENAI_KEY=${{ secrets.OPENAI_KEY }}
    --build-arg PROXY_USERNAME=${{ secrets.PROXY_USERNAME }}
    --build-arg PROXY_PASSWORD=${{ secrets.PROXY_PASSWORD }}
    -f Dockerfile .
```

I cannot build the image because of error I am encountering during workflow run.  It fails constantly whatever I do.
Replaced ">" with "|", added "\" slashes at the end it does not work

Here is the error message I always get:

```
Run docker build --tag api:latest --build-arg MONGODB_URL=*** --build-arg OPENAI_KEY=***  --build-arg PROXY_USERNAME=*** --build-arg PROXY_PASSWORD=*** -f Dockerfile-prod .
/home/runner/work/_temp/6c37fb37-33cd-[4](step:6:5)d05-96b9-a8d8624f236e.sh: line 1: --build-arg: command not found
ERROR: "docker buildx build" requires exactly 1 argument.
See 'docker buildx build --help'.

Usage:  docker buildx build [OPTIONS] PATH | URL | -

Start a build
Error: Process completed with exit code 127

```

I also tried to run with single long line `run` command. It still gives same error

**Comment by TrevorThomson** [comment, marked as answer]
Do your secrets contain any special characters? You might have to quote the --build-arg arguments...
Optionally, you could try using: https://github.com/docker/build-push-action

**Reply by serdarjan1995** [reply]
Seems to be there was special char that needs to be quoted.
Many thanks for the tip

**Comment by micky0001** [comment]
Same Issue also with me  - name: Build, tag, and push image to ECR
      run: |
        IMAGE_TAG=$(echo "$GITHUB_SHA" | cut -c1-7)
        echo "Building image with tag: $IMAGE_TAG"
        docker build -t "${{ secrets.ECR_REGISTRY }}/${{ secrets.ECR_REPOSITORY }}:$IMAGE_TAG" .
        docker push "${{ secrets.ECR_REGISTRY }}/${{ secrets.ECR_REPOSITORY }}:$IMAGE_TAG"
        echo "IMAGE_TAG=$IMAGE_TAG" >> $GITHUB_ENV


Getting error Run IMAGE_TAG=$(echo $GITHUB_SHA | cut -c1-7)
ERROR: docker: 'docker buildx build' requires 1 argument

Usage:  docker buildx build [OPTIONS] PATH | URL | -

Run 'docker buildx build --help' for more information
Error: Process completed with exit code 1.
